types/types.cc

```diff
@@ -1609,6 +1631,143 @@ static void validate_aux(const tuple_type_impl& t, View v) {
     }
 }
 
+sstring vector_type_impl::make_name(data_type type, size_t dimension) {
+    // To keep format compatibility with Origin we never wrap
+    // vector name into
+    // "org.apache.cassandra.db.marshal.FrozenType(...)".
+    return seastar::format("org.apache.cassandra.db.marshal.VectorType({}, {})", type->name(), dimension);
+}
+
+vector_type_impl::vector_type_impl(data_type elements, size_t dimension)
+        : concrete_type(kind::vector, make_name(elements, dimension),
+        elements->value_length_if_fixed() ? std::optional(elements->value_length_if_fixed().value()*dimension):std::nullopt),
+        _elements_type(elements), _dimension(dimension) {
+    _contains_set_or_map = _elements_type->contains_set_or_map();
+}
+
+shared_ptr<const vector_type_impl>
+vector_type_impl::get_instance(data_type elements, size_t dimension) {
+    return intern::get_instance(elements, dimension);
+}
+
+bytes_view read_vector_element(bytes_view& v, std::optional<size_t> value_length_if_fixed) {
```

Comment:
In the end the duplicated function wasn't needed, because the same result could be achieved by just creating a basic_single_fragmented_view from a bytes_view and using the version for fragmented views. I've added a commit that fixes this.